### ISSUE TITLE
v3.8.0-rc.1: bearer token based helm pull fails to list tags

### ISSUE THREAD

**Issue by sabre1041** [helm/helm#10559]
When attempting to perform a `helm pull` when targeting an OCI registry with token based authentication, an error occurs listing tags

```shell
./helm pull oci://testhelmoci.azurecr.io/helm-charts/test-oci --version 0.1.0
Error: GET "https://testhelmoci.azurecr.io/v2/helm-charts/test-oci/tags/list": GET "https://testhelmoci.azurecr.io/oauth2/token?scope=repository%3Ahelm-charts%2Ftest-oci%3Ametadata_read%2Cpull&service=testhelmoci.azurecr.io": unexpected status code 401: unauthorized: authentication required, visit https://aka.ms/acr/authorization for more information.
```

Helm Version: `v3.8.0-rc.1`

**Comment by kingdonb** [comment]
I used this approach to try to reproduce the issue so I can verify #10558, but I was not able to trigger this error on the current release. I'm using an azurecr.io container registry and I followed this guide, after struggling with Google container registries and IAM a while to try to get a similar config, I thought it might be easier to reproduce it on the exact same environment where the original report was from...

https://docs.microsoft.com/en-us/azure/container-registry/container-registry-helm-repos

There are three authentication methods described and it wasn't clear to me which one I should use to ensure a bearer token exchange through the headers. I'm a bit new when it comes to OCI charts, forgive me 😀 I went with this one:

```
USER_NAME="00000000-0000-0000-0000-000000000000"
PASSWORD=$(az acr login --name $ACR_NAME --expose-token --output tsv --query accessToken)
```

If you can give me a hint about how to arrange a repro of this issue, then I can help to verify #10558

**Comment by dhanesh012** [comment]
I am facing the same issue. 

@kingdonb , You should be getting an error when you try to pull a chart using `helm pull` without chart version specified.

**Comment by kingdonb** [comment]
OK, I have confirmed the fix 👍 at c137bfb6 I receive this error:

```
Error: GET "https://acrkingdontest.azurecr.io/v2/helm/hello-world/tags/list": GET "https://acrkingdontest.azurecr.io/oauth2/token?scope=repository%3Ahelm%2Fhello-world%3Ametadata_read%2Cpull&service=acrkingdontest.azurecr.io": unexpected status code 401: unauthorized: authentication required, visit https://aka.ms/acr/authorization for more information.
```

With the updated version at: `version.BuildInfo{Version:"v3.7+unreleased", GitCommit:"c8a2559236ff814e57b6a2f6076a2646f5bad130", GitTreeState:"clean", GoVersion:"go1.17.6"}` I get this result instead:

```
$ ~/bin/helm pull oci://acrkingdontest.azurecr.io/helm/hello-world
Pulled: acrkingdontest.azurecr.io/helm/hello-world:0.1.0
Digest: sha256:60d942de42296e029e603c92e540c705abab4a1b3a6fb77d81cc88356a1aead1
```

(This does not work at all in the previous released version, so I built a binary from the commit before this PR)

@scottrigby I was able to verify this one, so it looks good to merge 👍
